### PR TITLE
Refactoring StackName enum into a class with methods

### PR DESCRIPTION
**Pull request by tlisonbee**

### CODE REVIEW

File: src/main/java/com/nike/cerberus/domain/environment/StackName.java [hunk 1]
```diff
@@ -16,47 +16,94 @@
...
+    private static final String TEMPLATE_PATH_ROOT = "/cloudformation/";
+
+    private final Logger logger = LoggerFactory.getLogger(StackName.class);
+
 
     private final String name;
+    private final String templatePath;
 
-    StackName(final String name) {
+    private StackName(final String name, final String yamlFileName) {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/domain/environment/StackName.java]: nit: we should change `yamlFileName` to `cloudFormationFileName` or `cfFileName`

File: src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java [hunk 2]
```diff
@@ -191,7 +180,8 @@ public boolean isRunnable(final UpdateStackCommand command) {
         }
 
 
-        final TypeReference<Map<String, String>> typeReference = new TypeReference<Map<String, String>>() {};
+        final TypeReference<Map<String, String>> typeReference = new TypeReference<Map<String, String>>() {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java]: nit: I know IntelliJ autoformats the closing brace to the next line, but I've been putting them on the same line

File: src/main/java/com/nike/cerberus/util/StackNameConverter.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.nike.cerberus.util;
+
+import com.beust.jcommander.IStringConverter;
+import com.nike.cerberus.domain.environment.Stack;
+
+public class StackNameConverter implements IStringConverter<Stack> {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/util/StackNameConverter.java]: nit: should this be called `StackConverter`

File: src/main/java/com/nike/cerberus/domain/environment/Stack.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+    private static final String TEMPLATE_PATH_ROOT = "/cloudformation/";
+
+    private final Logger logger = LoggerFactory.getLogger(Stack.class);
+
+
+    private final String name;
+    private final String templatePath;
+
+    private Stack(final String name, final String yamlFileName) {
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/domain/environment/Stack.java]: nit: we should change `yamlFileName` to `cloudFormationFileName` or `cfFileName` or `templateFileName`

Comment (tlisonbee) [reply src/main/java/com/nike/cerberus/domain/environment/Stack.java]: done

File: src/main/java/com/nike/cerberus/util/StackNameConverter.java (hunk 3 above)
Comment (tlisonbee) [reply src/main/java/com/nike/cerberus/util/StackNameConverter.java]: Done

File: src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java (hunk 2 above)
Comment (tlisonbee) [reply src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java]: Done

**Pull request merged**